### PR TITLE
Add option to set workdays remaining hours automatically on entrydialog

### PR DESCRIPTION
**Pull request by kurukimi**
Preference toggle to select whether entry form should automatically set remaining hours of the day to the input

### CODE REVIEW

File: web/src/components/entry-dialog/useEntryForm.tsx [hunk 1]
```diff
@@ -69,6 +73,32 @@ const useEntryForm = ({ editEntry, date }: useEntryProps) => {
...
+      const totalDuration = totalDurationOfEntries(
+        data?.findWorkdays.length === 1 ? data.findWorkdays[0].entries : [],
+      );
+      const workDayFullHours = dayjs.duration(7.5, "hour");
+      const remainingHours =
+        workDayFullHours > totalDuration
+          ? workDayFullHours.subtract(totalDuration)
+          : dayjs.duration(0);
+      const remainingHoursFormatted = roundToFullMinutes(remainingHours).asHours().toString();
+      setDuration(remainingHoursFormatted);
```
Comment (kurukimi) [inline web/src/components/entry-dialog/useEntryForm.tsx:96]: IDK if reset [(reset)](https://react-hook-form.com/docs/useform/reset) would be better option to set the duration. I tried using reset quickly but it bugged somehow -> maybe I didn't know how to use it. Using reset we can set the default values of the form, which means that when user wants to clear the form manually, the duration would reset to that value.

Comment (kurukimi) [reply web/src/components/entry-dialog/useEntryForm.tsx]: Using reset now it should work as excpected

File: web/src/components/entry-dialog/EntryForm.tsx [hunk 2]
```diff
@@ -154,16 +164,36 @@ const EntryForm = ({ reset, onSubmit, editEntry, originalDate, form, loading }:
...
+                      }
+                      label={t("entryDialog.setRemainingHours")}
+                    />
+                  </FormGroup>
+                )}
+              </Box>
+            </Grid>
+            <Grid item xs={8} sx={{ mt: 2 }}>
+              <Box sx={{ display: "flex", justifyContent: "end", gap: 2 }}>
+                <Button type="reset" variant="outlined" size="large" onClick={() => reset()}>
```
Comment (kurukimi) [inline web/src/components/entry-dialog/EntryForm.tsx:188]: This () => reset() instead of just reset fixed the jittering

**Review by joonashak — APPROVED** [review]
👌

**Pull request merged**